plugins/logger/zap_adapter.go

```diff
@@ -17,20 +18,39 @@ func NewZapAdapter(zapLogger *zap.Logger) *ZapAdapter {
 	}
 }
 
+func separateFields(keyVals []interface{}) (fields []zap.Field, individualKeyVals []interface{}) {
+	for _, value := range keyVals {
```

Comment:
Hey @peterfox . Since logger is smt heavily used in the application, there is no reason to make extra allocations here. In the `range` statement when iterating over the `map` type, the second arg is a copy of the value. It's better here to use:
`keyVals[key]` to avoid extra copying.

Reply:
A good point to bring up, thank you for pointing it out. This should be fixed now.